### PR TITLE
Added OpenStack Support

### PR DESCRIPTION
**Pull request by afritzler**
- Support for OpenStack machine class
- `managevm` cli to test the various machine classes

### CODE REVIEW

File: pkg/driver/driver_openstack.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+		UserData:         []byte(d.UserData),
+		AvailabilityZone: availabilityZone,
+	}
+
+	createOpts = &keypairs.CreateOptsExt{
+		CreateOptsBuilder: createOpts,
+		KeyName:           keyName,
+	}
+
+	glog.Infof("creating machine")
```
Comment (prashanth26) [inline pkg/driver/driver_openstack.go]: Change log levels to V(3). Make all glog.Info* to glog.V(3).Info*

File: pkg/driver/driver_openstack.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	}
+
+	return openstack.NewComputeV2(client, gophercloud.EndpointOpts{
+		Region:       strings.TrimSpace(region),
+		Availability: gophercloud.AvailabilityPublic,
+	})
+}
+
+func (d *OpenStackDriver) encodeMachineId(instanceId string) string {
+	return fmt.Sprintf("openstack:///%s \n", instanceId)
```
Comment (prashanth26) [inline pkg/driver/driver_openstack.go]: Add region of the machine in instanceID on creation, similar to the case of AWS driver. Also please remove the trailing '\n' in the encode function.

**Comment by hardikdr** [comment]
Thanks for the PR.
@afritzler Please feel free to refrain yourself from doing heavy rebasing, as we had to merge other PR.
We will help you by creating PR on your branch for rebasing later. You might want to fix only the changes requested by prashanth.

**Comment by afritzler** [comment]
Let me factor in your requested changes. I will push a commit to this PR.

**Comment by afritzler** [comment]
Ok, I adjusted the issues you mentioned above with 2cffccf

**Comment by prashanth26** [comment]
Thanks @afritzler. It looks good now.  I shall resolve the merge conflicts. 

I wanted to test it out, but I am still having trouble with building the cloud-init for open-stack. Maybe we can sync up sometime this after noon when you are free.

**Comment by prashanth26** [comment]
Merge conflicts have been resolved. A new docker-image version has also been created.

**Pull request merged**